src/types.rs

```diff
@@ -582,6 +640,164 @@ impl ZeroSsl {
     }
 }
 
+/// A unique certificate identifier for the ACME renewal information (ARI) extension
+///
+/// See <https://www.ietf.org/archive/id/draft-ietf-acme-ari-07.html#section-4.1> for
+/// more information.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct CertificateIdentifier<'a> {
+    /// The BASE64URL-encoded authority key identifier (AKI) extension `keyIdentifier` of the certificate
+    pub authority_key_identifier: Cow<'a, str>,
+
+    /// The BASE64URL-encoded serial number of the certificate
+    pub serial: Cow<'a, str>,
+}
+
+impl CertificateIdentifier<'_> {
+    /// Encode a unique certificate identifier using the provided authority key ID and serial
+    ///
+    /// `authority_key_identifier` must be the DER-encoded ASN.1 octet string from the
+    /// `keyIdentifier` field of the `AuthorityKeyIdentifier` extension found in the certificate
+    /// to be identified.
+    ///
+    /// `serial` must be the DER-encoded ASN.1 serial number from the certificate to be identified.
+    /// Care must be taken to use the **encoded** serial number, not a big integer representation.
+    ///
+    /// The combination uniquely identifies a certificate within all certificates issued by the
+    /// same CA.
+    ///
+    /// See [RFC 5280 §4.1.2.2], [RFC 5280 §4.2.1.1], and [draft-ietf-acme-ari-07 §4.1]
+    ///
+    /// [RFC 5280 §4.1.2.2]: https://www.rfc-editor.org/rfc/rfc5280#section-4.1.2.2
+    /// [RFC 5280 §4.2.1.1]: https://www.rfc-editor.org/rfc/rfc5280#section-4.2.1.1
+    /// [draft-ietf-acme-ari-07 §4.1]: https://www.ietf.org/archive/id/draft-ietf-acme-ari-07.html#section-4.1
+    pub fn new(authority_key_identifier: Der<'_>, serial: Der<'_>) -> Self {
+        Self {
+            authority_key_identifier: BASE64_URL_SAFE_NO_PAD
+                .encode(authority_key_identifier)
+                .into(),
+            serial: BASE64_URL_SAFE_NO_PAD.encode(serial).into(),
+        }
+    }
+
+    /// Convert the `CertificateIdentifier` into an owned version with a static lifetime
+    pub fn into_owned(self) -> CertificateIdentifier<'static> {
+        CertificateIdentifier {
+            authority_key_identifier: Cow::Owned(self.authority_key_identifier.into_owned()),
```

Comment:
I think you can call `Cow::to_owned()` to make this a little neater.

Reply:
Writing:
```rust
    pub fn into_owned(self) -> CertificateIdentifier<'static> {
        CertificateIdentifier {
            authority_key_identifier: Cow::to_owned(&self.authority_key_identifier),
            serial:  Cow::to_owned(&self.serial),
        }
    }
```

gives a lifetime error. Am I missing something?

Reply:
Call it as a method on the field?

Reply:
You mean like this?

```rust
        CertificateIdentifier {
            authority_key_identifier: self.authority_key_identifier.to_owned(),
            serial: self.serial.to_owned(),
        }
``` 

Same result :thinking:

Reply:
I'm going to leave it as is, but if you want to take a crack at cleaning this up I'd be interested to see what I'm missing :laughing:

Reply:
Hmm, I think I had too high expectations of the `impl<T> ToOwned for Cow`...